### DISCUSSION TITLE
Result Parser Interceptor

### DISCUSSION THREAD

**Discussion by jpholtmeyer** [gajus/slonik#532] [Q&A]
I'm trying to use Slonik with Zod to validate query results.
It's a little unclear in the documentation if I need to implement my own custom Result Parser Interceptor for validation to happen.

https://github.com/gajus/slonik?tab=readme-ov-file#example-use-of-sqltype
It would appear that validation should work out-of-the-box here.
 
https://github.com/gajus/slonik?tab=readme-ov-file#runtime-validation
And according to step 2 here, I would need to create my own interceptor.

Does anyone know if a custom interceptor is required?

**Comment by firxworx** [comment]
This is covered by the docs here (which is not one of your links above so you may have missed it): 
https://github.com/gajus/slonik#user-content-result-parser-interceptor

In your client configuration object (type `ClientConfiguration`) in the `interceptors` array is where you put the interceptor.

Once this is in place then you can use `sql.type(zodSchemaGoesHere)` to specify a zod schema.

If your query returns many results and you are using `pool.many(...)` to query it you only need to specify zod schema of a single item (same as `pool.one(...)`) not of an array of the items.

```ts
const result = await pool.many(sql.type(zProductDto.strict())`
      SELECT ... ;
    `)
```

See the README section on errors for the types of errors thrown in different cases including where result validation fails.
